superset/utils/async_query_manager.py

```diff
@@ -0,0 +1,200 @@
+            or config["DATA_CACHE_CONFIG"]["CACHE_TYPE"] == "null"
+        ):
+            raise Exception(
+                """
+                Cache backends (CACHE_CONFIG, DATA_CACHE_CONFIG) must be configured
+                and non-null in order to enable async queries
+                """
+            )
+
+        if len(config["GLOBAL_ASYNC_QUERIES_JWT_SECRET"]) < 32:
+            raise AsyncQueryTokenException(
+                "Please provide a JWT secret at least 32 bytes long"
+            )
+
+        self._redis = redis.Redis(  # type: ignore
+            **config["GLOBAL_ASYNC_QUERIES_REDIS_CONFIG"], decode_responses=True
+        )
+        self._stream_prefix = config["GLOBAL_ASYNC_QUERIES_REDIS_STREAM_PREFIX"]
+        self._stream_limit = config["GLOBAL_ASYNC_QUERIES_REDIS_STREAM_LIMIT"]
+        self._stream_limit_firehose = config[
+            "GLOBAL_ASYNC_QUERIES_REDIS_STREAM_LIMIT_FIREHOSE"
+        ]
+        self._jwt_cookie_name = config["GLOBAL_ASYNC_QUERIES_JWT_COOKIE_NAME"]
+        self._jwt_cookie_secure = config["GLOBAL_ASYNC_QUERIES_JWT_COOKIE_SECURE"]
+        self._jwt_secret = config["GLOBAL_ASYNC_QUERIES_JWT_SECRET"]
+
+        @app.after_request
+        def validate_session(  # pylint: disable=unused-variable
+            response: Response,
+        ) -> Response:
+            reset_token = False
+            user_id = session["user_id"] if "user_id" in session else None
+
+            if "async_channel_id" not in session or "async_user_id" not in session:
+                reset_token = True
+            elif user_id != session["async_user_id"]:
+                reset_token = True
+
+            if reset_token:
+                async_channel_id = str(uuid.uuid4())
+                session["async_channel_id"] = async_channel_id
+                session["async_user_id"] = user_id
+
+                token = self.generate_jwt(
+                    {"channel": async_channel_id, "user_id": user_id}
+                )
+
+                response.set_cookie(
+                    self._jwt_cookie_name,
+                    value=token,
+                    httponly=True,
+                    secure=self._jwt_cookie_secure,
+                    # max_age=max_age or config.cookie_max_age,
+                    # domain=config.cookie_domain,
+                    # path=config.access_cookie_path,
+                    # samesite=config.cookie_samesite
+                )
+
+            return response
+
+    def generate_jwt(self, data: Dict[str, Any]) -> str:
+        encoded_jwt = jwt.encode(data, self._jwt_secret, algorithm="HS256")
+        return encoded_jwt.decode("utf-8")
+
+    def parse_jwt(self, token: str) -> Dict[str, Any]:
+        data = jwt.decode(token, self._jwt_secret, algorithms=["HS256"])
+        return data
+
+    def parse_jwt_from_request(self, request: Request) -> Dict[str, Any]:
+        token = request.cookies.get(self._jwt_cookie_name)
```

Comment:
Again, I'd delegate this to the lib. One other case that's missing here is `Authorization` header support

Reply:
See comments above re: `FLASK_JWT_EXTENDED`